### PR TITLE
`D300`: prevent autofix when both triples are in body

### PR DESCRIPTION
**Pull request by T-256**
## Summary
Addresses https://github.com/astral-sh/ruff/issues/8402#issuecomment-1788782750

## Test Plan

Added associated test

### CODE REVIEW

File: crates/ruff_linter/src/rules/pydocstyle/rules/triple_quotes.rs [hunk 1]
```diff
@@ -61,12 +61,12 @@ impl Violation for TripleSingleQuotes {
...
     let leading_quote = docstring.leading_quote();
 
-    let prefixes = docstring
-        .leading_quote()
+    let prefixes = leading_quote
         .trim_end_matches(|c| c == '\'' || c == '"')
         .to_owned();
 
     let expected_quote = if docstring.body().contains("\"\"\"") {
+        if docstring.body().contains("\'\'\'") { return }
```
Comment (T-256) [inline crates/ruff_linter/src/rules/pydocstyle/rules/triple_quotes.rs:69]:
```suggestion
        if docstring.body().contains("\'\'\'") {
            return;
        }
```

**Comment by T-256** [comment]
Sorry for disabling auto-merge, but I think this should have autofix:
```py
def contains_triples(t):
    '''(\'''|\""")'''
```
Into
```py
def contains_triples(t):
    """(\'''|\""")"""
```

**Comment by charliermarsh** [comment]
I think this is fine to merge as-is because we already get that "wrong" today for the simpler case, I think:

```python
def contains_triples(t):
    '''(\""")'''
```

Getting it right requires that we parse the entire docstring to track escapes.

**Comment by T-256** [comment]
> Getting it right requires that we parse the entire docstring to track escapes

Correct, I was looking for escape-processed of docstring content to implement it, but didn't find a proper way.
BTW I added a todo example, I also think it's safe to merge for now.

**Comment by charliermarsh** [comment]
@T-256 - We have some stuff like that in `crates/ruff_linter/src/rules/flake8_quotes/rules/avoidable_escaped_quote.rs`.

**Pull request merged**